src/vm/checker/typecheck/contexts.rs

```diff
@@ -64,6 +64,45 @@ impl ContractAnalysis {
             .expect(SERIALIZE_FAIL_MESSAGE)
     }
 
+    pub fn to_interface(&self) -> interface::ContractInterface {
```

Comment:
Just a note to myself, and maybe a note to add to a PR check-list:  If you add any new kinds of objects to the Clarity VM, you'll also need to update this method so that it will generate a `ContractInterface` that represents everything accurately.

Reply:
This could be forced via destructuring the self reference:

```
let Self{ private_function_types, public_function_types, read_only_function_types, variable_types, persisted_variable_types, map_types } = self;
```

And adding a comment that any new contract objects must be added to the contract interface as well.

Reply:
Sounds good.  Let's do this before merging -- it would make it hard to accidentally omit data from the contract interface.

Reply:
Implemented